### DISCUSSION TITLE
How can I use a njk include in an md template

### DISCUSSION THREAD

**Discussion by mwinterheller** [11ty/eleventy#3245] [Uncategorized]
Hi,
really love eleventy! Thank you so much for this powerful tool.

I have a markdown blog post which I save as an .md file. Inside I would like to use a njk include.

_{% include 'get-download-element.njk' %}_

This worked great and the element got rendered within the blog post without issues. However, now I have to actually use njk logic within the *get-download-element.njk* template.  Everything after the line where the logic starts within the _get-download-element.njk_  seems to gets escaped? I can partly overcome this by not using any indents - but this does not seem to work too well either.

I am sure I am missing something here.

Thank you for the help!
/Michael

**Comment by edheltzel** [comment]
@mwinterheller in your eleventy configuration is where you configure where 11ty will look for includes - [see docs](https://www.11ty.dev/docs/config/#directory-for-includes)

so in your example `{% include 'get-download-element.njk' %}` would be in the root of your `_includes` if you did not set a custom directory, which would live inside of your `src` (again unless you set a custom directory)

**Reply by mwinterheller** [reply]
Hi @edheltzel - thank you for the swift reply and help. However, I do not have any issue with using the include itself. This works just fine.

The content of the include is something like:
<img width="516" alt="Code Example" src="https://github.com/11ty/eleventy/assets/17781197/26981408-f6fe-4d74-a680-fdb6381c7b68">

I marked the lines that get included (rendered as HTML) fine and those that don't within the including markdown post. This results in the following render result:

<img width="607" alt="Render Example" src="https://github.com/11ty/eleventy/assets/17781197/3c35ab31-b03e-4051-87c0-b51a48fe6c71">

Does this make more sense?

Thank you,
/Michael

**Reply by pdehaan** [reply]
Unfortunately, Markdown is very whitespace sensitive, so I'm guessing its getting converted to a `<pre>` block due to indenting.
You might be able to use something like https://mozilla.github.io/nunjucks/templating.html#whitespace-control

Although i thought that feature was disabled by default in Eleventy@2+, per https://www.11ty.dev/docs/languages/markdown/#indented-code-blocks

**Reply by edheltzel** [reply]
@mwinterheller did you try setting `markdownTemplateEngine: "njk",` in your eleventy config? 

I use macros in my markdown files for several projects, which are no different than includes with logic. 
```handlebars
---
title: Get $20 off your next brake pad and rotor repair service
permalink: /20-off-brakes/
---
{% from "components/button/macro.njk" import button %}

## Sign up to get $20 off your next Brake Pad and/or Rotor Replacement Service. {.uk-text-center}

  {{ button({
    color: 'secondary',
    icon: 'triangle-right',
    size: 'large',
    width: 'uk-width-1-1',
    url: '/contact/schedule-appointment/',
    label: 'Book your appointment today!'
  })}}
```

**Reply by mwinterheller** [reply]
@edheltzel & @pdehaan thank you guys! Really appreciate the swift help here! Unfortunately, disabeling the white space setting did not bring about any change. And we already have the **markdownTemplateEngine: "njk"**  setting set. 

Only by removing the logic from the include I can get the include to render properly.

I will try to not use the include but create a macro and use that option just like in your example @edheltzel next.

Thank you,
/Michael

**Reply by edheltzel** [reply]
@mwinterheller You could try adding the `| safe` filter to your if

**Reply by mwinterheller** [reply]
@edheltzel  Thank you! So I tried **{% if locale === 'fr' | safe %}** but it it did not seem to make a difference. 

I also tried setting **templateEngineOverride: njk, md** in the post blog, but it also did not change the rendering.

**Reply by edheltzel** [reply]
What happens if you change the file type to .njk

**Reply by mwinterheller** [reply]
So I tried it as macro - does not seem to make a difference.

I tried changing the file type of the MD to .njk -> now the include renders fine but the markdown formatting is lost.

**Comment by edheltzel** [comment]
> Although i thought that feature was disabled by default in Eleventy@2+, per https://www.11ty.dev/docs/languages/markdown/#indented-code-blocks
@pdehaan Yeah you're correct on that.

**Comment by mwinterheller** [comment]
Thank you everyone who tried to help! I ended up removing all whitespaces and adding additional line breaks. That worked well enough.

**Reply by pdehaan** [reply]
Another option i sometimes use is just make .njk (or .liquid) templates, and render partial Markdown snippets using the https://www.11ty.dev/docs/plugins/render/ tags. May or may not work for your use case, but it is an option if you want to avoid worrying about whitespace control within Nunjucks snippets.
